src/opencl/mod.rs

```diff
@@ -88,25 +89,75 @@ impl<T> Buffer<T> {
     }
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, Hash)]
+pub struct DeviceUuid([u8; utils::CL_UUID_SIZE_KHR]);
+
+impl TryInto<DeviceUuid> for &str {
+    type Error = String;
```

Comment:
This should be a new error type, e.g. `Uuid(String)` where `String` is the original input, and you could then create an error message like: `Cannot parse UUID, expected one formatted as aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee, got {0}.`